<launch>

    <rosparam file="$(find tank_setup)/config/tank_model_$(optenv TANK_MODEL)_params.yaml" command="load" />
    <rosparam file="$(find tank_setup)/config/tank_motor_model_$(optenv TANK_MOTOR_MODEL)_params.yaml" command="load" />

    <node name="tank_agent_node" pkg="tank_agent" type="tank_agent_node" output="screen">
        <param name="serialPort" type="string" value="/dev/ttyS0" />
        <param name="serialBaudrate" type="string" value="115200" />
        <param name="imuMsgPublishEnabled" type="bool" value="true" />
        <param name="magMsgPublishEnabled" type="bool" value="true" />
        <param name="odomImuMsgPublishEnabled" type="bool" value="true" />
        <param name="odomEncoderMsgPublishEnabled" type="bool" value="true" />
        <param name="odomMsgPublishEnabled" type="bool" value="false" />
        <param name="odomTfBroadcastEnabled" type="bool" value="false" />
<!--
        <param name="logTankMsg" type="string" value="00000010" />
        <param name="logTankCmd" type="string" value="00000001" />
-->
    </node>

    <node name="ekf_localization_node" pkg="robot_localization" type="ekf_localization_node" clear_params="true" output="screen">
        <rosparam command="load" file="$(find tank_setup)/config/robot_localization_ekf.yaml" />
        <remap from="odometry/filtered" to="odom" />
    </node>

    <node name="tank_setup_tf_node" pkg="tank_setup" type="tank_setup_tf_node" output="screen" />

    <node name="rplidarNode" pkg="rplidar_ros" type="rplidarNode" output="screen">
        <remap from="scan" to="rplidar_scan" />

        <param name="serial_port" type="string" value="/dev/ttyUSB0" />
        <param name="serial_baudrate" type="int" value="115200" />
        <param name="frame_id" type="string" value="base_laser" />
        <!-- scan_mode: Legacy, Express, Boost -->
        <param name="scan_mode" type="string" value="Boost" /> 
        <param name="inverted" type="bool" value="false" />
        <param name="angle_compensate" type="bool" value="true" />
    </node>

    <node name="laser_filter" pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen">
        <remap from="scan" to="rplidar_scan" />
        <remap from="scan_filtered" to="scan" />

        <rosparam command="load" file="$(find tank_setup)/config/laser_filter_config.yaml" />
    </node>

</launch>
